### ISSUE TITLE
dnsmasq catching SIGTERM repeatedly on OpenWRT 22.03.2

### ISSUE THREAD

**Issue by clvgt12** [openwrt/openwrt#11006]
Router configuration:
```

Release 22.03.2 is currently running
A current stable release available for download is 22.03.2
A current stable release available for download is 21.02.5
OpenWRT runtime environment: openwrt ath79 generic
Hardware environment:
	Model ID=tplink_archer-a7-v5
	Model Name=tp-link_archer_a7_v5
	Board=tplink_archer-a7-v5
```
Packages installed:
```
#opkg list-installed
at - 3.2.2-1
ath10k-board-qca988x - 20220411-1
ath10k-firmware-qca988x-ct - 2020-11-08-1
base-files - 1494-r19803-9a599fee93
busybox - 1.35.0-4
ca-bundle - 20211016-1
ca-certificates - 20211016-1
cgi-io - 2022-08-10-901b0f04-21
ddns-scripts - 2.8.2-25
ddns-scripts-services - 2.8.2-25
dnsmasq - 2.86-14
dropbear - 2022.82-2
firewall4 - 2022-10-14-4fbf6d75-1
fstools - 2022-06-02-93369be0-2
fwtool - 2019-11-12-8f7fe925-1
getrandom - 2021-08-03-205defb5-2
hostapd-common - 2022-01-16-cff80b4f-13.1
https-dns-proxy - 2022-10-15-1
iw - 5.16-1
iwinfo - 2022-08-19-0dad3e66-1
jansson4 - 2.13.1-2
jq - 1.6-2
jshn - 2022-05-15-d2223ef9-1
jsonfilter - 2018-02-04-c7e938d6-1
kernel - 5.10.146-1-72e39b50629eaf33fc9600ab703ee2ab
kmod-ath - 5.10.146+5.15.58-1-1
kmod-ath10k-ct - 5.10.146+2022-05-13-f808496f-1
kmod-ath9k - 5.10.146+5.15.58-1-1
kmod-ath9k-common - 5.10.146+5.15.58-1-1
kmod-cfg80211 - 5.10.146+5.15.58-1-1
kmod-crypto-aead - 5.10.146-1
kmod-crypto-ccm - 5.10.146-1
kmod-crypto-cmac - 5.10.146-1
kmod-crypto-crc32c - 5.10.146-1
kmod-crypto-ctr - 5.10.146-1
kmod-crypto-gcm - 5.10.146-1
kmod-crypto-gf128 - 5.10.146-1
kmod-crypto-ghash - 5.10.146-1
kmod-crypto-hash - 5.10.146-1
kmod-crypto-hmac - 5.10.146-1
kmod-crypto-manager - 5.10.146-1
kmod-crypto-null - 5.10.146-1
kmod-crypto-rng - 5.10.146-1
kmod-crypto-seqiv - 5.10.146-1
kmod-crypto-sha256 - 5.10.146-1
kmod-gpio-button-hotplug - 5.10.146-3
kmod-hwmon-core - 5.10.146-1
kmod-lib-crc-ccitt - 5.10.146-1
kmod-lib-crc32c - 5.10.146-1
kmod-mac80211 - 5.10.146+5.15.58-1-1
kmod-nf-conntrack - 5.10.146-1
kmod-nf-conntrack6 - 5.10.146-1
kmod-nf-flow - 5.10.146-1
kmod-nf-log - 5.10.146-1
kmod-nf-log6 - 5.10.146-1
kmod-nf-nat - 5.10.146-1
kmod-nf-reject - 5.10.146-1
kmod-nf-reject6 - 5.10.146-1
kmod-nfnetlink - 5.10.146-1
kmod-nft-core - 5.10.146-1
kmod-nft-fib - 5.10.146-1
kmod-nft-nat - 5.10.146-1
kmod-nft-offload - 5.10.146-1
kmod-nls-base - 5.10.146-1
kmod-phy-ath79-usb - 5.10.146-1
kmod-ppp - 5.10.146-1
kmod-pppoe - 5.10.146-1
kmod-pppox - 5.10.146-1
kmod-slhc - 5.10.146-1
kmod-usb-core - 5.10.146-1
kmod-usb-ehci - 5.10.146-1
kmod-usb-ledtrig-usbport - 5.10.146-1
kmod-usb2 - 5.10.146-1
libblobmsg-json20220515 - 2022-05-15-d2223ef9-1
libc - 1.2.3-4
libcares - 1.18.1-1
libcurl4 - 7.85.0-6.1
libelf1 - 0.186-1
libev - 4.33-1
libgcc1 - 11.2.0-4
libiwinfo-data - 2022-08-19-0dad3e66-1
libiwinfo-lua - 2022-08-19-0dad3e66-1
libiwinfo20210430 - 2022-08-19-0dad3e66-1
libjson-c5 - 0.15-2
libjson-script20220515 - 2022-05-15-d2223ef9-1
liblua5.1.5 - 5.1.5-10
liblucihttp-lua - 2022-07-08-6e68a106-1
liblucihttp0 - 2022-07-08-6e68a106-1
libmbedtls12 - 2.28.1-1
libmnl0 - 1.0.5-1
libnftnl11 - 1.2.1-1
libnghttp2-14 - 1.44.0-1
libnl-tiny1 - 2021-11-21-8e0555fb-1
libpthread - 1.2.3-4
libubox20220515 - 2022-05-15-d2223ef9-1
libubus-lua - 2022-06-01-2bebf93c-1
libubus20220601 - 2022-06-01-2bebf93c-1
libuci20130104 - 2021-10-22-f84f49f0-6
libuclient20201210 - 2021-05-14-6a6011df-1
libucode20220812 - 2022-10-07-4ae70721-1
libustream-wolfssl20201210 - 2022-01-16-868fd881-2
libwolfssl5.5.1.ee39414e - 5.5.1-stable-3
logd - 2021-08-03-205defb5-2
lua - 5.1.5-10
luci - git-20.074.84698-ead5e81
luci-app-ddns - git-22.285.40043-938bb8f
luci-app-firewall - git-22.089.67563-7e3c1b4
luci-app-https-dns-proxy - 2021-11-22-7
luci-app-opkg - git-22.273.29004-9f6876b
luci-base - git-22.288.45147-96ec0cd
luci-compat - git-22.069.45071-03bb0e2
luci-lib-base - git-20.232.39649-1f6dc29
luci-lib-ip - git-20.250.76529-62505bd
luci-lib-ipkg - git-18.318.71164-4bbe325
luci-lib-jsonc - git-22.097.61921-7513345
luci-lib-nixio - git-20.234.06894-c4a4e43
luci-mod-admin-full - git-19.253.48496-3f93650
luci-mod-network - git-22.244.54818-b13d8c7
luci-mod-status - git-22.189.48501-6731190
luci-mod-system - git-22.264.46172-b6b7da4
luci-proto-ipv6 - git-21.148.48881-79947af
luci-proto-ppp - git-21.158.38888-88b9d84
luci-ssl - git-20.244.36115-e10f954
luci-theme-bootstrap - git-22.288.45147-96ec0cd
mtd - 26
netifd - 2022-08-25-76d2d41b-1
nftables-json - 1.0.2-2.1
odhcp6c - 2022-08-05-7d21e8d8-18
odhcpd-ipv6only - 2022-03-22-860ca900-1
openwrt-keyring - 2022-03-25-62471e69-3
opkg - 2022-02-24-d038e5b6-1
ppp - 2.4.9.git-2021-01-04-3
ppp-mod-pppoe - 2.4.9.git-2021-01-04-3
procd - 2022-06-01-7a009685-1
procd-seccomp - 2022-06-01-7a009685-1
procd-ujail - 2022-06-01-7a009685-1
px5g-wolfssl - 5.1
rpcd - 2022-09-21-8c852b65-1
rpcd-mod-file - 2022-09-21-8c852b65-1
rpcd-mod-iwinfo - 2022-09-21-8c852b65-1
rpcd-mod-luci - 20210614
rpcd-mod-rrdns - 20170710
swconfig - 12
uboot-envtools - 2022.01-32
ubox - 2021-08-03-205defb5-2
ubus - 2022-06-01-2bebf93c-1
ubusd - 2022-06-01-2bebf93c-1
uci - 2021-10-22-f84f49f0-6
uclient-fetch - 2021-05-14-6a6011df-1
ucode - 2022-10-07-4ae70721-1
ucode-mod-fs - 2022-10-07-4ae70721-1
ucode-mod-ubus - 2022-10-07-4ae70721-1
ucode-mod-uci - 2022-10-07-4ae70721-1
uhttpd - 2022-08-12-e3395cd9-1
uhttpd-mod-ubus - 2022-08-12-e3395cd9-1
urandom-seed - 3
urngd - 2020-01-21-c7f7b6b6-1
usign - 2020-05-23-f1f65026-1
wireless-regdb - 2022.08.12-1
wpad-basic-wolfssl - 2022-01-16-cff80b4f-13.1
zlib - 1.2.11-6
```
Running processes:  (NOTE, dnsmasq and https-dns-proxy are being respawned continuously after SIGTERM signal is raised)
```
# ps
  PID USER       VSZ STAT COMMAND
    1 root      1704 S    /sbin/procd
    2 root         0 SW   [kthreadd]
    3 root         0 IW   [kworker/0:0-eve]
    4 root         0 IW<  [kworker/0:0H-kb]
    6 root         0 IW<  [mm_percpu_wq]
    7 root         0 SW   [rcu_tasks_trace]
    8 root         0 SW   [ksoftirqd/0]
    9 root         0 IW<  [netns]
   42 root         0 SW   [oom_reaper]
  121 root         0 IW<  [writeback]
  123 root         0 SW   [kcompactd0]
  135 root         0 IW<  [kblockd]
  136 root         0 IW<  [blkcg_punt_bio]
  137 root         0 SW   [watchdogd]
  171 root         0 SW   [kswapd0]
  173 root         0 IW<  [kthrotld]
  217 root         0 SW   [spi0]
  408 root         0 IW<  [ipv6_addrconf]
  427 root         0 SW   [irq/14-keys]
  428 root         0 SW   [irq/15-keys]
  546 root         0 SWN  [jffs2_gcd_mtd5]
  655 ubus      1336 S    /sbin/ubusd
  656 root       996 S    /sbin/askfirst /usr/libexec/login.sh
  691 root      1108 S    /sbin/urngd
  778 root         0 IW<  [cfg80211]
  816 root         0 IW<  [ath10k_wq]
  817 root         0 IW<  [ath10k_aux_wq]
  818 root         0 IW<  [ath10k_tx_compl]
 1465 root      2276 S    /sbin/rpcd -s /var/run/ubus/ubus.sock -t 30
 1775 root      2656 S    {hostapd} /sbin/ujail -t 5 -n hostapd -U network -G network -C /etc/capabilities/wpad.json -c -- /usr/sbin/hostapd -s -g /var/run/hostapd
 1776 root      2656 S    {wpa_supplicant} /sbin/ujail -t 5 -n wpa_supplicant -U network -G network -C /etc/capabilities/wpad.json -c -- /usr/sbin/wpa_supplicant -
 1779 network   4512 S    /usr/sbin/hostapd -s -g /var/run/hostapd/global
 1802 network   4424 S    /usr/sbin/wpa_supplicant -n -s -g /var/run/wpa_supplicant/global
 1842 root      1824 S    /sbin/netifd
 1900 root      1648 S    /usr/sbin/odhcpd
 2021 root      4176 S    /usr/sbin/uhttpd -f -h /www -r tomato -x /cgi-bin -u /ubus -t 60 -T 30 -k 20 -A 1 -n 3 -N 100 -R -p 0.0.0.0:80 -p [::]:80 -C /etc/uhttpd.
 2456 root      1308 S    udhcpc -p /var/run/udhcpc-eth0.2.pid -s /lib/netifd/dhcp.script -f -t 0 -i eth0.2 -x hostname:tomato -C -R -O 121
 2457 root      1124 S    odhcp6c -s /lib/netifd/dhcpv6.script -P0 -t120 eth0.2
 2499 root      2656 S    {ntpd} /sbin/ujail -t 5 -n ntpd -U ntp -G ntp -C /etc/capabilities/ntpd.json -c -u -r /bin/ubus -r /usr/bin/env -r /usr/bin/jshn -r /usr/
 2507 ntp       1308 S    /usr/sbin/ntpd -n -N -S /usr/sbin/ntpd-hotplug -p 0.openwrt.pool.ntp.org -p 1.openwrt.pool.ntp.org -p 2.openwrt.pool.ntp.org -p 3.openwrt
 2722 root      1208 S    /usr/sbin/dropbear -F -P /var/run/dropbear.1.pid -s -g -p 192.168.1.1:22 -p fdb4:eac3:2ca2::1:22 -K 300 -T 3
 2758 root         0 IW<  [kworker/0:2H-ev]
 3369 root      1228 S    /usr/sbin/dropbear -F -P /var/run/dropbear.1.pid -s -g -p 192.168.1.1:22 -p fdb4:eac3:2ca2::1:22 -K 300 -T 3 -2 9
 3377 root      1312 S    -ash
 5930 nobody    1080 S    /usr/sbin/atd -f
 6171 logd      1324 S    /sbin/logd -S 64
10672 root      1228 S    /usr/sbin/dropbear -F -P /var/run/dropbear.1.pid -s -g -p 192.168.1.1:22 -p fdb4:eac3:2ca2::1:22 -K 300 -T 3 -2 9
10673 root      1312 S    -ash
14322 root         0 IW   [kworker/u2:2-ev]
15900 root         0 IW   [kworker/0:1-eve]
28133 root         0 IW   [kworker/u2:1-ev]
28270 root      2656 S    {dnsmasq} /sbin/ujail -t 5 -n dnsmasq -u -l -r /bin/ubus -r /etc/TZ -r /etc/dnsmasq.conf -r /etc/ethers -r /etc/group -r /etc/hosts -r /e
28275 dnsmasq   1492 S    /usr/sbin/dnsmasq -C /var/etc/dnsmasq.conf.cfg01411c -k -x /var/run/dnsmasq/dnsmasq.cfg01411c.pid
28312 nobody    3084 S    /usr/sbin/https-dns-proxy -r https://family.cloudflare-dns.com/dns-query -a 127.0.0.1 -p 5054 -b 1.1.1.3,1.0.0.3 -4 -u nobody -g nogroup
28549 root      1308 S    sleep 15
28557 root      1124 S    odhcp6c -s /lib/netifd/dhcpv6.script -P0 -t120 eth0.2
28558 root      1316 R    ps
30889 root      1228 S    /usr/sbin/dropbear -F -P /var/run/dropbear.1.pid -s -g -p 192.168.1.1:22 -p fdb4:eac3:2ca2::1:22 -K 300 -T 3 -2 9
30890 root      1312 S    -ash
32220 root         0 IW   [kworker/u2:0-ev]
```
dnsmasq repeatedly catching SIGTERM signals

```
# logread|grep SIGTERM
Tue Oct 18 10:20:31 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:21 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:26 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:36 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:42 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:51 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:21:56 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:22:06 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:22:11 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:22:19 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:22:25 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:23:00 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:23:05 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:23:14 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:23:19 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:06 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:11 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:19 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:25 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:35 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:40 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:48 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 10:25:53 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
root@tomato:~/scripts# date
Tue Oct 18 10:26:05 EDT 2022
```

**Comment by clvgt12** [comment]
Update,  dnsmasq is also catching SIGTERM repeatedly on Release 22.03.0.   Note the https-dns-proxy package is loaded and configured.

```
# do_sysupgrade.sh 
Usage: /root/scripts/do_sysupgrade.sh [RELEASE_LABEL]
Release 22.03.0 is currently running
A current stable release available for download is 22.03.2
A current stable release available for download is 21.02.5
OpenWRT runtime environment: openwrt ath79 generic
Hardware environment:
	Model ID=tplink_archer-a7-v5
	Model Name=tp-link_archer_a7_v5
	Board=tplink_archer-a7-v5
root@tomato:~/scripts# logread|grep dnsmasq | grep SIGTERM
Tue Oct 18 12:57:17 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:23 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:29 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:34 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:40 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:37 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:43 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
root@tomato:~/scripts# date
Tue Oct 18 12:58:51 EDT 2022
root@tomato:~/scripts# date
Tue Oct 18 12:59:11 EDT 2022
root@tomato:~/scripts# logread|grep dnsmasq | grep SIGTERM
Tue Oct 18 12:57:17 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:23 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:29 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:34 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:40 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:37 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:43 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:52 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:57 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:05 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:10 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
root@tomato:~/scripts# logread|grep dnsmasq | grep SIGTERM
Tue Oct 18 12:57:17 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:23 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:29 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:34 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:57:40 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:37 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:43 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:52 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:58:57 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:05 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:10 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:22 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:27 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:35 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Oct 18 12:59:40 2022 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
```

**Comment by ynezz** [comment]
> Note the https-dns-proxy package is loaded and configured.

This package comes from [packages feed](https://github.com/openwrt/packages), there is [separate issue tracker](https://github.com/openwrt/packages) there.
